.github/workflows/ci.yml

```diff
@@ -414,6 +415,51 @@ jobs:
       - name: Run via package Cargo.toml
         run: cargo run semver-checks check-release --manifest-path subject/sqllogictest/Cargo.toml
 
+  run-on-tokio:
+    # cargo-semver-checks crashed here due to improper CLI argument handling:
+    # https://github.com/obi1kenobi/cargo-semver-checks/issues/380
+    name: Run cargo-semver-checks on tokio ~v1.25.0
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout cargo-semver-checks
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Checkout tokio
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          repository: 'tokio-rs/tokio'
+          ref: 'd7b7c6131774ab631be6529fef3680abfeeb4781'
+          path: 'subject'
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v2
+        with:
+          key: tokio
+
+      # This test caught a bug where the default baseline was set to the current path,
+      # instead of the default registry version.
+      - name: Run semver-checks on tokio crate manifest only
+        run: cargo run semver-checks check-release --manifest-path="subject/tokio/Cargo.toml" --release-type minor --exclude benches --exclude examples --exclude stress-test --exclude tests-build --exclude tests-integration
+
+      # This test caught a bug where `--exclude` was silently ignored
+      # if `--workspace` wasn't set at the same time.
+      - name: Run semver-checks on workspace manifest with explicit exclusions
+        run: cargo run semver-checks check-release --manifest-path="subject/Cargo.toml" --release-type minor --exclude examples --exclude stress-test --exclude tests-build --exclude tests-integration
+
+      # This test caught a bug where `publish = false` items in a workspace were semver-checked
+      # unless either explicit `--workspace` was present or was implied e.g. via `--exclude`.
+      - name: Run semver-checks on workspace manifest with implicit exclusions
+        run: cargo run semver-checks check-release --manifest-path="subject/Cargo.toml" --release-type minor
```

Comment:
I'm not 100% sure I understand the expected behavior of `--workspace` in relation to `--exclude` and `publish = false` crates.

If my understanding is correct, all these commands are expected to pass. The `benches / examples / <etc.>` crates are all `publish = false` and `subject/Cargo.toml` is the workspace manifest file which doesn't define any packages by itself.

@epage could you take a look at let me know if my interpretation here makes sense?